ddtrace/contrib/vertica/patch.py

```diff
@@ -0,0 +1,246 @@
+                    },
+                    "copy": {
+                        "operation_name": "vertica.copy",
+                        "span_type": "vertica",
+                        "on_before": copy_before,
+                    },
+                    "fetchone": {
+                        "operation_name": "vertica.fetchone",
+                        "span_type": "vertica",
+                        "on_after": fetch_after,
+                    },
+                    "fetchall": {
+                        "operation_name": "vertica.fetchall",
+                        "span_type": "vertica",
+                        "on_after": fetch_after,
+                    },
+                    "nextset": {
+                        "operation_name": "vertica.nextset",
+                        "span_type": "vertica",
+                        "on_after": fetch_after,
+                    },
+                },
+            },
+        },
+    },
+)
+
+def patch():
+    global _PATCHED
+    if _PATCHED:
+        return
+
+    _install(config.vertica)
+    _PATCHED = True
+
+
+def unpatch():
+    global _PATCHED
+    if _PATCHED:
+        _uninstall(config.vertica)
+        _PATCHED = False
+
+def _uninstall(config):
+    for patch_class_path in config["patch"]:
+        patch_mod = '.'.join(patch_class_path.split('.')[0:-1])
+        patch_class = patch_class_path.split('.')[-1]
+
+        for patch_routine in config["patch"][patch_class_path]["routines"]:
+            mod = importlib.import_module(patch_mod)
+            cls = getattr(mod, patch_class)
+            unwrap(cls, patch_routine)
+
+def _install(config):
+    for patch_class_path in config["patch"]:
+        patch_mod = '.'.join(patch_class_path.split('.')[0:-1])
+        patch_class = patch_class_path.split('.')[-1]
+
+        for patch_routine in config["patch"][patch_class_path]["routines"]:
+            def _wrap():
+                # _wrap is needed to provide data to the wrappers which can
+                # only be provided from a function closure.
+
+                _patch_routine = patch_routine
+                _patch_item = patch_class_path
+
+                # patch the __init__ of the class with a Pin instance containing the defaults
+                @wrapt.patch_function_wrapper(patch_mod, "{}.{}".format(patch_class, "__init__"))
+                def init_wrapper(wrapped, instance, args, kwargs):
+                    wrapped(*args, **kwargs)
+
+                    # create and attach a pin with the defaults
+                    Pin(
+                        service=config["service_name"],
+                        app=config["app"],
+                        app_type=config["app_type"],
+                        tags=config.get("tags", {}),
+                        tracer=config.get("tracer", ddtrace.tracer),
+                        _config=config["patch"][_patch_item],
+                    ).onto(instance)
+
+                def _find_config(instance, routine_name):
```

Comment:
This is temporary/a placeholder until a better solution is thought up. This code is not run in the vertica integration since the situation in which it is needed (described in the long comment below) does not occur in the tracing configuration for the vertica library.